### ISSUE TITLE
[Regression] XMB Stopped Working (#14090)

### ISSUE THREAD

**Issue by Jonathan44062** [RPCS3/rpcs3#14123]
### Quick summary

The famous PR which fixes GT Games breaks XMB, It has taken us a while to realize 
It crashes with this error message
`F {PPU[0x100002b] Thread (ScePafJob) [mms: 0x00d19b5c]} VM: Access violation reading location 0x1c (unmapped memory)`

### Details

Working 0.0.27-15096
![image](https://github.com/RPCS3/rpcs3/assets/78843189/7f586df1-0e3f-4b4e-950d-bb82aa5f80ab)

Broken 0.0.27-15111 (https://github.com/RPCS3/rpcs3/pull/13917)
![image](https://github.com/RPCS3/rpcs3/assets/78843189/cadded07-bfe8-4839-ae06-0c7153aa0ca8)

### Build with regression

0.0.27-15111

### Attach two log files

Working
[XMB Works.log.gz](https://github.com/RPCS3/rpcs3/files/11932064/XMB.Works.log.gz)

Broken
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11932078/RPCS3.log.gz)

### Attach capture files for visual issues

None

### System configuration

OS - Windows 10 22H2
CPU - i7 8700
GPU - GTX NIVIDIA 1070 Ti
Driver version - 536.23

**Comment by Darkhost1999** [comment]
![image](https://github.com/RPCS3/rpcs3/assets/60384196/4d77daa7-b995-4742-93c6-221ab87b1629)
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11932124/RPCS3.log)
I have not been able to reproduce this with ryzen 5 3600 zenver 2, gtx 1660 super 536.23 on windows 11 insider.
I tried clearing caches, building new ones. I tried booting without precompiling, and with precompiling. I don't know how to make it crash

**Comment by azsenca** [comment]
Same for me. Only black screen... Ryzen 5 5500, rtx 3060ti, windows 11

Here is the log: 
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11945802/RPCS3.log)

**Comment by Tha14** [comment]
I have the same hardware as @Darkhost1999 but I am on Windows 10. For me it crashes early on:
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11973705/RPCS3.log)

**Comment by MastaG** [comment]
For me it crashes immediately on Fedora 38.

**Comment by AniLeo** [comment]
@elad335 can you check as it comes from #13917?

**Comment by elad335** [comment]
Enable "PPU Calling History" in vsh config file and upload log. I can't reproduce locally.

**Comment by Jonathan44062** [comment]
Still crashes, I set PPU Calling History to True
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11984641/RPCS3.log)

**Comment by letsgoawaydev** [comment]
happening to me with 
AMD Ryzen 5 5600X
RTX 3060ti

**Comment by letsgoawaydev** [comment]
it might be related to the nvidia drivers, as everybody here seems to have nvidia cards.

**Comment by elad335** [comment]
Relared debug improvements wete merged, it's a good idea to re-upload log with PPU Calling History enabled.

**Comment by Jonathan44062** [comment]
New log
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11999378/RPCS3.log.gz)

**Comment by letsgoawaydev** [comment]
also this actually broke a game for me, the game is called nfl blitz. it wont move past the loading screen. using the version where the xmb works fixes this issue

**Comment by solarmystic** [comment]
I discovered something that could be of interest (maybe). 

If I compile all the VSH PPU modules on build 0.0.28-15111 (the affected build according this issue report (after wiping all existing VSH PPU cache in `rpcs3\cache` of course), VSH will work just fine on rpcs3, at least on my end.

![image](https://github.com/RPCS3/rpcs3/assets/4345150/5923273c-394c-46ab-80e3-8f5df2fe9f96)

I could then bring forward those compiled modules to boot VSH with no issue even on the latest master. (0.0.28-15341 as of this post)

![image](https://github.com/RPCS3/rpcs3/assets/4345150/6856086b-e12c-4ab7-85e9-986f797cb48f)

On the other hand, if I compile VSH PPU modules on 0.0.28-15096 (again, making sure i wipe all PPU cache in `rpcs3\cache`), I can't bring those modules forward and it will then cause the exact issue as stated in this issue report. 

![image](https://github.com/RPCS3/rpcs3/assets/4345150/aa327846-e803-45f5-a75f-a52c856b3d50)

Logfiles:-
Not working if PPU modules compiled from 15096 brought forward
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/12013025/RPCS3.zip)

Working if PPU modules compiled from 15111 brought forward
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12013165/RPCS3.log.gz)

**Comment by demerest** [comment]
Have the same issue in 15341
Wiping cache didn't help
Ryzen 7 5700X
RTX 4070

**Comment by jlnhlfan** [comment]
I can't even get it to load because new modules get endlessly created just when you think they've all been created, or it looks like it's about to be completed.

Processor	Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz   2.90 GHz
Installed RAM	16.0 GB
GPU processor:		NVIDIA GeForce GTX 1650

**Comment by Lillowh** [comment]
Same issue here when loading xmb
F {PPU[0x100002b] Thread (ScePafJob) [mms: 0x00d19b5c]} VM: Access violation reading location 0x1c (unmapped memory)

I9 i5-990k @5.0 GHz
Installed Ram: 32.0GB
GPU processor: NVIDIA GeForce RTX 3080

**Comment by ieee802dot11ac** [comment]
Can confirm I am also having this issue on latest master. Arch Linux, R5 3600, Radeon RX 570, 16GB ram. Updating to latest firmware (4.90) did not help. 

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12208616/RPCS3.log.gz)

**Comment by Darkhost1999** [comment]
> Enable "PPU Calling History" in vsh config file and upload log. I can't reproduce locally.

Delete your mms folder and rebuild your database on next boot of vsh.
It's not vsh that's broken its the mms contents.

**Comment by elad335** [comment]
Can the issue be safely closed than?

**Comment by Darkhost1999** [comment]
> Can the issue be safely closed than?

No, because it's still a regression. The database should be able to complete being built. I'm just saying that vsh can't display anything because none of it's database can be built after the mentioned build. Vsh.self works fine.

**Comment by Darkhost1999** [comment]
You can find crash logs in the dev_hdd0/mms/db folder. They're written in the actual PS3 format as if from a real PS3.

**Comment by elad335** [comment]
Can you bitwise compare which files are different when produced by the the two builds?

**Comment by Darkhost1999** [comment]
Sadly, I'm not sure how to do that. That's why I'm hoping you can reproduce by moving your mms folder somewhere else then booting vsh.self to watch it fail

**Comment by Darkhost1999** [comment]
> Delete your mms folder and rebuild your database on next boot of vsh.
> It's not vsh that's broken its the mms contents.

These instructions aren't working for me today. The issue is still produced in a clean folder of RPCS3 but in my personal folder deleting the mms folder rebuilds the database and restarts vsh just fine. Booting and working on master for me.

**Comment by Darkhost1999** [comment]
#14090
[rpcs3-v0.0.28-15255-b0a34670_win64-works.log](https://github.com/RPCS3/rpcs3/files/12254754/RPCS3.log)
<img width="1280" alt="Screenshot 2023-08-03 151737" src="https://github.com/RPCS3/rpcs3/assets/60384196/2d1a6804-e26a-4e6b-b14b-c0917c7e672e">
[rpcs3-v0.0.28-15259-554b27a8_win6-broken.log](https://github.com/RPCS3/rpcs3/files/12254718/RPCS3.log)
<img width="1280" alt="Screenshot 2023-08-03 151516" src="https://github.com/RPCS3/rpcs3/assets/60384196/76d3b4fa-8bc3-4025-b714-2958a36bb9c5">
Broken folder does not produce an mms folder.
<img width="429" alt="Screenshot 2023-08-03 152253" src="https://github.com/RPCS3/rpcs3/assets/60384196/2a075548-31d5-4936-9a38-04a70f890d76">
Reason the regression went unnoticed is that those who have their mms folder from before the regression can keep booting fine without issue. As noted I managed to reproduce before by deleting the mms folder. However, on master it seems that that is no longer the case. A file that is different after a certain build is this db.err file. The build that allows the production of this db.err file is assumed to be from the same regression.
[db.zip](https://github.com/RPCS3/rpcs3/files/12254791/db.zip)
Compressed to be able to upload. Size of file is 1kb.
I was asking Elad if they know how to look at this file because it is an error as if produced by a real ps3
![image](https://github.com/RPCS3/rpcs3/assets/60384196/c0ee8af9-356e-4d4d-8522-32945c24a064)
So basically if you do rebuild your database on master branch this is the file that is different from before and this is what I can see from that error file.
<img width="229" alt="Screenshot 2023-08-03 151120" src="https://github.com/RPCS3/rpcs3/assets/60384196/eae46662-c68a-4edc-91bd-57974ff44b4d">
I performed a lot of testing to come to this conclusion

**Comment by azsenca** [comment]
@Darkhost1999 Can you upload your mms folder, so I can try if it works for me? I have mms folder, but it is empty...

**Comment by Darkhost1999** [comment]
> @Darkhost1999 Can you upload your mms folder, so I can try if it works for me? I have mms folder, but it is empty...

The mms from your ps3 should work. Or just downgrade your rpcs3 version and build 1. I don't know the copyright for the files, so I do not wish to accidentally violate them.

**Comment by TheKurtix** [comment]
Same problem

F {PPU[0x100002b] Thread (ScePafJob) [mms: 0x00d19b5c]} VM: Access violation reading location 0x1c (unmapped memory)

Win 11 - Ryzen 7 3700x - Nvidia 3080

**Comment by soulcollectivex** [comment]
Very similar issue happens with me running RPCS3 Version "0.0.29-15444-6672499d Alpha | master"
I have not been able to view the XMB at all, it just complies ppu modules and then stays at a black screen.
there are multiple errors in the log, but I also get this error:
F {PPU[0x100002b] Thread (ScePafJob) [mms: 0x00d19b5c]} VM: Access violation reading location 0x1c (unmapped memory)

I'm running RPCS3 on a Razer Blade 15 Advanced Edition Laptop with Intel 11th gen core i7 11800H, Nvidia RTX 3080 Laptop GPU, with 64gb of ram

**Comment by the-loan-wolf** [comment]
i am also having this same problem with rpcs3-v0.0.29-15452-343ba873_linux64.AppImage
`qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""

WARNING: radv is not a conformant Vulkan implementation, testing use only.

WARNING: radv is not a conformant Vulkan implementation, testing use only.

Failed to initialize QMediaPlayer "Could not find the videoconvert GStreamer element"

Failed to initialize QAudioOutput "Could not find the audioconvert GStreamer element"

RPCS3: PPU[0x100002b] Thread (ScePafJob) [mms: 0x00d19b5c]: VM: Access violation reading location 0x1c (unmapped memory)`

**Comment by elad335** [comment]
For it to work locally you need frash cache such as from a fresh install, there is no "remove BVSH cache" yet.

**Comment by MSuih** [comment]
On windows, you can erase all folders not marked with game serial from `cache` directory to clear both firmware and VSH modules. Then start VSH again and both will be recompiled.

**Comment by MastaG** [comment]
On Linux (Fedora 38) with todays git and starting fresh (removing ~/.config/rpcs3 and ~/.cache/rpcs3) it seems to be fixed :)
Vsh now compiles and runs fine.
